sycl/include/CL/sycl/detail/cg_types.hpp

```diff
@@ -169,39 +169,43 @@ constexpr bool isKernelLambdaCallableWithKernelHandlerImpl() {
                                         kernel_handler>();
 }
 
-// Type traits to find out if kernal lambda has kernel_handler argument
+// Type trait to find out if kernal lambda has kernel_handler argument
+template <typename KernelType, typename LambdaArgType = void>
+struct KernelLambdaHasKernelHandlerArgT {
+  constexpr static bool value =
+      isKernelLambdaCallableWithKernelHandlerImpl<KernelType, LambdaArgType>();
+};
+
+// Helpers for running kernel lambda on the host device
 
 template <typename KernelType>
-constexpr bool isKernelLambdaCallableWithKernelHandler() {
-  return check_kernel_lambda_takes_args<KernelType, kernel_handler>();
+typename detail::enable_if_t<
+    KernelLambdaHasKernelHandlerArgT<KernelType>::value, void>
```

Comment:
Not a request to change, just minor comment. std::enable_if_t is available in c++14. This could be written a bit shorter:
```suggestion
std::enable_if_t<KernelLambdaHasKernelHandlerArgT<KernelType>::value>
```